modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java

```diff
@@ -224,8 +224,11 @@ public void apply(Project project) {
 
 									@Override
 									public void execute(Task task1) {
-										_validateClientExtension(
-											clientExtension, project);
+										ValidateClientExtensionTask validateClientExtensionTask = new ValidateClientExtensionTask(
```

Comment:
It seems like `ValidateClientExtensionTask` is just functioning as a util or helper. Is there a reason it needs to be a dedicated task class? Could we not just add an additional helper method inline?

Reply:
yeah, just moved out to a dedicated task class to improve legibility and modularity . This `ClientExtensionProjectConfigurator` seems to be inflated with a lot of code. Since there is a package for tasks, I   just extracted for one there. Does it make sense?

Reply:
Sure, that makes sense. And while I absolutely agree that there are opportunities to make the class a little leaner, there are a few problems with the current approach:

1. It is confusing to mix business logic changes and a refactoring into a single commit, since it obfuscates what is happening. Each change should be made separately.
2. There is unnecessary indirection here. If we are going to create a new class called `ValidateClientExtensionTask`, it should be used as a task directly instead of being a delegate for the current task.
3. Adding a new task to that package will cause it to become public API and configurable in the groovy DSL. I am not certain that is what we want at this point.

For simplicity's sake, I think we should add the logic inline for now and explore shrinking the class separately.

Reply:
Oh cool, makes sense